src/main/java/org/alfresco/utility/data/DataContent.java

```diff
@@ -189,57 +303,139 @@ public String addEmailAlias(String alias) throws Exception
     }
 
     /**
-     * Creates a random document based on {@link DocumentType} passed
-     * Return the {@link Document} object on success creation
+     * It will create a new file in current resource
      * <code>
-     * dataContent.usingUser(userModel).usingResource(myFolder).createContent(DocumentType.TEXT_PLAIN);
-     * </code>
-     * 
-     * @param documentType
+     * FileModel newRandomFile = FileModel.getRandomFileModel();
+     * dataContent.usingUser(testUser).usingSite(testSite).createContent(newRandomFile);
+     * <code>
+     */
+    public FileModel createContent(){
+        return createContent(new FileModel(RandomData.getRandomName("File")));
+    }
+
+    /**
+     * This is the entry point of the createContent() method to make REST API or CMIS call
+     */
+    public FileModel createContent(FileModel fileModel){
+        AlfrescoHttpClient client = alfrescoHttpClientFactory.getObject();
+
+        if(client.getAlfVersion() >= 5.2)
+        {
+            FileModel createFile = createContentV1Api(client, fileModel);
+            this.updateContent(client, createFile);
+            return createFile;
+        }
+        else
+        {
+            return createContentCmisApi(fileModel);
+        }
+    }
+
+    /**
+     * Creates a file using REST API
+     * Return the {@link FileModel} object on success creation
+     *
+     * @param client
+     * @param fileModel
      * @return
-     * @throws DataPreparationException
      */
-    public FileModel createContent(DocumentType documentType) throws DataPreparationException
+    public FileModel createContentV1Api(AlfrescoHttpClient client, FileModel fileModel)
     {
-        String newContent = String.format("%s.%s", RandomData.getRandomName("file"), Utility.cmisDocTypeToExtentions(documentType));
-        String newLocation = Utility.buildPath(getLastResource(), newContent);
-        STEP(String.format("DATAPREP: Creating a new non-empty content %s in %s ", newContent, getLastResource()));
+        // Build request
+        String nodeId = this.getNodeRef();
+        String reqUrl = client.getApiVersionUrl() + "nodes/" + nodeId + "/children";
+        HttpPost post  = new HttpPost(reqUrl);
+        JSONObject body = new JSONObject();
+        body.put("name", fileModel.getName());
+        body.put("nodeType", "cm:content");
 
-        if (getLastResource().isEmpty())
-            setLastResource(RandomData.getRandomName("Folder"));
+        // Set Title or Description if specified
+        if (fileModel.getTitle() != null || fileModel.getDescription() != null)

```

Comment:
shouldn't this be 2 "if"s ? 
is it ok to pass one of those parameters with value null ?
Should we also check if title is not empty? is an empty title a valid property ?

Reply:
Changed